### PR TITLE
feat(registrar): enabled HTML in email notifications

### PR DESCRIPTION
**Pull request by HejdaJakub**
* Now is possible to set also html notification template (not only plain text).
* If is set html template, then will be send both html and plain text messages (html or plain text message will be displayed according to client settings).

BREAKING CHANGE: update DB

### CODE REVIEW

**Comment by HejdaJakub** [comment]
My tests in [RegistrarBaseIntegrationTest.java](https://github.com/CESNET/perun/pull/3621/files#diff-98f736abea1037ccb22a5779a49edabd61e9058cea6c49d430834d84c4873340) are maybe more unit tests than integration. But at least for review, these tests can be useful.

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/MailManagerImpl.java [hunk 1]
```diff
@@ -1821,7 +1861,12 @@ private void mailValidation(Application app, ApplicationMail mail, List<Applicat
 					}
 
 					// set replaced text
-					message.setText(mailText);
+					if (containsHtmlMessage(mail, lang)) {
+						String alternativePlainText = getMailAlternativePlainText(mail, lang, app, data, reason, exceptions);
```
[the remark this answers is not in the document]
Comment (HejdaJakub) [reply perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/MailManagerImpl.java:1794]: Good point. I moved the block of code for checking and replacing `validationLink` and `redirectUrl` to the separate method (`replaceValidationLinkAndRedirectUrl(...)`) and I use it at the previous place and newly also here for checking/replacing alternativePlainText.

**Comment by HejdaJakub** [comment]
> Good job. One use-case came to my mind - what if a user accidentally sets HTML message to contain only whitespace - then he thinks it is not set, and only plain text will be used. But in fact, the blank HTML will be sent (because it is not technically empty). So, maybe we could check whether HTML text is blank (contains whitespace) instead of just empty. What do you think? Could this happen?

You're right. I replaced all occurreces of `isEmpty()` checks for html message by `isBlank()` checks.

**Review by cuadradek — APPROVED** [review]

**Comment by HejdaJakub** [comment]
I changed the charset in the `setContent()` method for `MimeBodyPart` to charset=utf-8 because the default charset doesn't display Czech letters correctly. 
image
After this change, it looks good.
image

**Review by cuadradek — APPROVED** [review]

**Review by Johaney-s — APPROVED** [review]

**Pull request merged**